torch/csrc/jit/pybind_utils.h

```diff
@@ -4,26 +4,66 @@
 
 namespace torch { namespace jit {
 
-inline Stack createStack(const py::tuple& tuple, size_t reserve_extra_space = 0) {
+inline Stack createStack(const py::tuple& tuple, at::ArrayRef<Value*> inputs, size_t reserve_extra_space = 0) {
+  if (tuple.size() != inputs.size()) {
+    throw std::runtime_error("expected " + std::to_string(inputs.size()) +
+                             " inputs, but got " + std::to_string(tuple.size()));
+  }
+  static const auto castToIValue = [](const py::object& obj, Type& t) -> IValue{
+    switch (t.kind()) {
+      case TypeKind::DynamicType:
+      case TypeKind::TensorType:
+        return py::cast<autograd::Variable>(obj);
+      case TypeKind::FloatType:
+        return py::cast<double>(obj);
+      case TypeKind::IntType:
+        return py::cast<int64_t>(obj);
+      case TypeKind::ListType:
+      case TypeKind::TupleType:
+        throw std::runtime_error("Lists and tuples are not supported yet");
+      case TypeKind::NumberType:
+        throw std::runtime_error("Insufficient type information to convert input");
+    }
+    throw std::runtime_error("Missing cases in castToIValue! File a bug report.");
+  };
   Stack result;
   result.reserve(tuple.size() + reserve_extra_space);
-  for(auto e : tuple) {
-    result.push_back(py::cast<autograd::Variable>(e));
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    result.push_back(castToIValue(tuple[i], *inputs[i]->type()));
   }
   return result;
 }
 
-inline py::object wrapStack(Stack&& outputs) {
+inline py::object wrapStack(Stack&& outputs, at::ArrayRef<Value*> output_vals) {
+  if (outputs.size() != output_vals.size()) {
+    throw std::runtime_error("expected " + std::to_string(output_vals.size()) +
+                             " outputs, but got " + std::to_string(outputs.size()));
+  }
+  static const auto createOutput = [](IValue && ivalue, Value * value) -> py::object {
+    switch (value->type()->kind()) {
+      case TypeKind::DynamicType:
+      case TypeKind::TensorType:
+        return py::cast(autograd::Variable(ivalue.toTensor()));
```

Comment:
nit: `std::move(ivalue).toTensor()` to transfer without refcount

Reply:
It's already an rvalue in here (see the argument type)